src/conditions.ts

```
@@ -35,7 +35,7 @@ export class ConditionMonitor extends EventEmitter {
       );
     } else if (condition instanceof ContractCondition) {
       await this.retry(
-        () => this.startMonitoringContract(condition),
+        () => this.startMonitoringContract(condition.contractAddress),
```

This call will fail as `contractAddress` is a `0x${string}` and `startMonitoringContract` expects something like a `ContractCondition`
The error is hidden now because `startMonitoringContract` signature expects `any` but it will throw when trying to run this line `const { contractAddress, abi, eventName, providerURL } = condition;`